composer/loggers/progress_bar_logger.py

```diff
@@ -21,44 +21,67 @@
 _IS_TRAIN_TO_KEYS_TO_LOG = {True: ['loss/train'], False: ['metrics/eval/Accuracy']}
 
 
-@dataclass
-class _ProgressBarLoggerInstanceState:
-    total: Optional[int]
-    description: str
-    position: int
-    keys_to_log: List[str]
-    n: int
-    epoch_metrics: Dict[str, Any]
+class _ProgressBar:
 
-
-class _ProgressBarLoggerInstance:
-
-    def __init__(self, file: TextIO, state: _ProgressBarLoggerInstanceState) -> None:
-        self.state = state
+    def __init__(
+        self,
+        total: Optional[int],
+        position: int,
+        bar_format: str,
+        file: TextIO,
+        metrics: Dict[str, Any],
+        keys_to_log: List[str],
+        unit: TimeUnit = TimeUnit.EPOCH,
+        epoch_style: bool = False,
+    ) -> None:
+        self.keys_to_log = keys_to_log
+        self.metrics = metrics
+        self.position = position
+        self.unit = unit
+        self.epoch_style = epoch_style
         self.pbar = tqdm.auto.tqdm(
-            total=state.total,
-            position=state.position,
-            # Putting state.description in bar_format to avoid floating colons.
-            bar_format=f'{state.description} {{l_bar}}{{bar:25}}{{r_bar}}{{bar:-1b}}',
+            total=total,
+            position=position,
+            bar_format=bar_format,
             file=file,
             dynamic_ncols=True,
+            postfix=metrics,
+            unit=unit.value,
         )
-        self.pbar.set_postfix(state.epoch_metrics)
 
     def log_data(self, data: Dict[str, Any]):
-        formatted_data = {k: format_log_data_value(v) for (k, v) in data.items() if k in self.state.keys_to_log}
-        self.state.epoch_metrics.update(formatted_data)
-        self.pbar.set_postfix(self.state.epoch_metrics)
+        formatted_data = {k: format_log_data_value(v) for (k, v) in data.items() if k in self.keys_to_log}
+        self.metrics.update(formatted_data)
+        self.pbar.set_postfix(self.metrics)
+
+    def update(self, n=1):
```

Comment:
nit: do we need the default value, or is it always specified? 
```suggestion
    def update(self, n: int):
```

Reply:
To match the `tqdm.update` API as closed as possible, choose to have the default.